addons/sales_team/models/crm_team_member.py

```diff
@@ -0,0 +1,188 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, exceptions, fields, models, _
+
+
+class CrmTeamMember(models.Model):
+    _name = 'crm.team.member'
+    _inherit = ['mail.thread']
+    _description = 'Sales Team Member'
+    _rec_name = 'user_id'
+    _order = 'create_date ASC'
+    _check_company_auto = True
+
+    crm_team_id = fields.Many2one(
+        'crm.team', string='Sales Team',
+        default=False,  # TDE: temporary fix to activate depending computed fields
+        check_company=True, index=True, required=True)
+    user_id = fields.Many2one(
+        'res.users', string='Salesperson',  # TDE FIXME check responsible field
+        check_company=True, index=True, ondelete='cascade', required=True,
+        domain="[('share', '=', False), ('id', 'not in', user_in_teams_ids), ('company_ids', 'in', user_company_ids)]")
+    user_in_teams_ids = fields.Many2many(
+        'res.users', compute='_compute_user_in_teams_ids',
+        help='UX: Give users not to add in the currently chosen team to avoid duplicates')
+    user_company_ids = fields.Many2many(
+        'res.company', compute='_compute_user_company_ids',
+        help='UX: Limit to team company or all if no company')
+    active = fields.Boolean(string='Active', default=True)
+    is_membership_multi = fields.Boolean(
+        'Multiple Memberships Allowed', compute='_compute_is_membership_multi',
+        help='If True, users may belong to several sales teams. Otherwise membership is limited to a single sales team.')
+    member_warning = fields.Text(compute='_compute_member_warning')
+    # salesman information
+    image_1920 = fields.Image("Image", related="user_id.image_1920", max_width=1920, max_height=1920)
+    image_128 = fields.Image("Image (128)", related="user_id.image_128", max_width=128, max_height=128)
+    name = fields.Char(string='Name', related='user_id.display_name', readonly=False)
+    email = fields.Char(string='Email', related='user_id.email')
+    phone = fields.Char(string='Phone', related='user_id.phone')
+    mobile = fields.Char(string='Mobile', related='user_id.mobile')
+    company_id = fields.Many2one('res.company', string='Company', related='user_id.company_id')
+
+    @api.constrains('crm_team_id', 'user_id', 'active')
+    def _constrains_membership(self):
+        # In mono membership mode: check crm_team_id / user_id is unique for active
+        # memberships. Inactive memberships can create duplicate pairs which is whyy
+        # we don't use a SQL constraint. Include "self" in search in case we use create
+        # multi with duplicated user / team pairs in it. Use an explicit active leaf
+        # in domain as we may have an active_test in context that would break computation
+        existing = self.env['crm.team.member'].search([
+            ('crm_team_id', 'in', self.crm_team_id.ids),
+            ('user_id', 'in', self.user_id.ids),
+            ('active', '=', True)
+        ])
```

Comment:
Wondering if you should just search_count for every membership in self and raise on a case by case basis?
Here it just complicates the code a bit and you might end up fetching relationships you don't care about since you use "in" for both fields.